### PR TITLE
Split out glue library as pulp-glue package

### LINKED ISSUE #628: [TASK] Split glue layer as a library into separate python package
https://discourse.pulpproject.org/t/rfc-reunite-the-library-that-drives-pulp-cli-and-squeezer/262

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #628

TODO Checklist:
- [x] Move the contexts of all plugins to glue.
- [x] Figure out the release process for the split packages.

Review Checklist:
- [ ] An issue is properly linked. [feature and bugfix only]
- [ ] Tests are present or not feasible.
- [ ] Commits are split in a logical way (not historically).

### CODE REVIEW

File: pulp-glue/pulp_glue/common/openapi.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def __init__(
             raise OpenAPIError(_("Cert is required if key is set."))
         self._session.headers.update(
             {
-                "User-Agent": user_agent or "Pulp-CLI openapi parser",
+                "User-Agent": user_agent or f"Pulp-glue openapi parser ({__version__})",
```
Comment (ggainey) [inline pulp-glue/pulp_glue/common/openapi.py:77]: :+1:

File: pulpcore/cli/core/user.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import click
+from pulp_glue.common.context import PluginRequirement  # PulpEntityContext,; pass_entity_context,
```
Comment (ggainey) [inline pulpcore/cli/core/user.py:2]: Is there a reason to leave the commented-out imports here?

**Review by ggainey — COMMENTED** [review]
This looks good! I have one question below. Also - should we give the Build Gang some advance warning on this? (assuming you haven't already)

File: pulpcore/cli/core/user.py (hunk 2 above)
Comment (mdellweg) [reply pulpcore/cli/core/user.py:2]: Not at all. TBH they were there before and i used sed to update _all_ the files.

File: setup.py [hunk 3]
```diff
@@ -39,12 +39,11 @@
     package_data={"": ["py.typed", "locale/*/LC_MESSAGES/*.mo"]},
     python_requires=">=3.6",
     install_requires=[
+        "pulp-glue>=0.17.0.dev,<0.18",
         "click>=8.0.0,<9.0.0",
-        "packaging",
```
Comment (ggainey) [inline setup.py:43]: So contexts were/are the only things that rely on 'packaging' and 'requests', and that's now all in pulp_glue, correct?

File: setup.py [hunk 4]
```diff
@@ -39,12 +39,11 @@
     package_data={"": ["py.typed", "locale/*/LC_MESSAGES/*.mo"]},
     python_requires=">=3.6",
     install_requires=[
+        "pulp-glue>=0.17.0.dev,<0.18",
```
Comment (ggainey) [inline setup.py:42]: The concern I have here is that we'll need a pulp-cli release when we release a new pulp-glue. That may be OK - I'm just trying to envision the release-process for the tooling as we go forward. Thoughts?

**Review by ggainey — COMMENTED** [review]
I left a couple of comments, but I think this all makes sense - I may need to see it "in action" to really have a feeling for how releasing is going to look.

File: setup.py (hunk 4 above)
Comment (mdellweg) [reply setup.py:42]: And bump2version will take care of it, i hope...

File: pulpcore/cli/common/context.py [hunk 5]
```diff
@@ -1,833 +1,23 @@
...
-    "%Y-%m-%dT%H:%M:%S",  # ... with time and T as a separator
-    "%y/%m/%d",  # US format
-    "%y/%m/%d %h:%M:%S %p",  # ... with time
-    "%y/%m/%dT%h:%M:%S %p",  # ... with time and T as a separator
-    "%y/%m/%d %H:%M:%S",  # ... with time 24h
-    "%y/%m/%dT%H:%M:%S",  # ... with time 24h and T as a separator
-    "%x",  # local format
-    "%x %X",  # ... with time
-    "%xT%X",  # ... with time and T as a separator
+from pulp_glue.common.context import (  # noqa: F401
```
Comment (ggainey) [inline pulpcore/cli/common/context.py:1]: This is so that any cli-plugins we don't know about, will continue working post-this-change, correct?

**Review by ggainey — COMMENTED** [review]
Left one question, but I think we're about ready to merge.

File: pulpcore/cli/common/context.py (hunk 5 above)
Comment (mdellweg) [reply pulpcore/cli/common/context.py:1]: Absolutey!

**Review by ggainey — APPROVED** [review]

**Pull request merged**